### PR TITLE
Add handling transaction history

### PR DESCRIPTION
**Pull request by toddkao**
transaction history

### CODE REVIEW

**Comment by toddkao** [comment]
> We shouldn't store the `status`. We should store the tx hashes every transactions. I suggest to create a custom hook to check the status. We should show status from the api, not storage.

@codingki 
We do already store the tx hashes (and other details of the transaction)
I was thinking about this a bit yesterday, here's what I'm thinking. We store the status for the transaction, but if there are any transactions in the history that are not completed (in some pending state) we can use the API to fetch the status in the background and then once they are completed, update the status. The logic is that once a transactions has completed, whether it did so successfully or failed the status is basically static and doesn't change anymore, but if pending we can use the API to fetch changes to the status.

**Comment by codingki** [comment]
> > We shouldn't store the `status`. We should store the tx hashes every transactions. I suggest to create a custom hook to check the status. We should show status from the api, not storage.
> 
> @codingki We do already store the tx hashes (and other details of the transaction) I was thinking about this a bit yesterday, here's what I'm thinking. We store the status for the transaction, but if there are any transactions in the history that are not completed (in some pending state) we can use the API to fetch the status in the background and then once they are completed, update the status. The logic is that once a transactions has completed, whether it did so successfully or failed the status is basically static and doesn't change anymore, but if pending we can use the API to fetch changes to the status.

@toddkao  right, that works. so only if the status pending we fetch

File: packages/widget-v2/src/state/swapExecutionPage.ts [hunk 1]
```diff
@@ -38,9 +40,10 @@ export const chainAddressesAtom = atom<Record<number, ChainAddress>>({});
 export const swapExecutionStateAtom = atomWithStorage<SwapExecutionState>(
   "swapExecutionState",
   {
-    route: undefined,
+    route: {} as RouteResponse,
```
Comment (codingki) [inline packages/widget-v2/src/state/swapExecutionPage.ts:43]: I think it's ok to make it undefined, any reason to do this?

Comment (toddkao) [reply packages/widget-v2/src/state/swapExecutionPage.ts:43]: yeah i forget why i did this tbh, I think having it be undefined makes sense

**Review by codingki — APPROVED** [review]

**Pull request merged**